token/services/network/common/rws/translator/translator.go

```diff
@@ -202,12 +202,20 @@ func (w *Translator) checkIssue(issue IssueAction) error {
 }
 
 func (w *Translator) checkTransfer(t TransferAction) error {
-	keys, err := t.GetInputs()
+	inputs, err := t.GetInputs()
 	if err != nil {
 		return errors.Wrapf(err, "invalid transfer: failed getting input IDs")
 	}
 	if !t.IsGraphHiding() {
-		for _, key := range keys {
+		rwsetKeys := make([]string, len(inputs))
```

Comment:
do we need this slice? we could merge the two for loops right?

Reply:
but a better solution might be
```
var keys []string
if t.IsGraphHiding() {
 keys = t.GetSerialNumbers()
} else {
 keys = make([]string, len(inputs))
 ...
}

for _, key := range keys {
 bytes, err := w.RWSet.GetState(w.namespace, key)
 if err != nil {
    return errors.Wrapf(err, "invalid transfer: failed getting state [%s]", key)
 }
 if len(bytes) != 0 {
  return errors.Errorf("invalid transfer: input is already spent [%s:%v]", key, bytes)
 }
}
```

Reply:
okay